src/Generator/Passes/MarshalStructAsValTypeIfPossiblePass.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using CppSharp.AST;
+
+namespace CppSharp.Passes
+{
+    public class MarshalStructAsValTypeIfPossiblePass : TranslationUnitPass
+    {
+        
+        public override bool VisitClassDecl(Class @class)
+        {
+            @class.Type = CheckClassIsStructible(@class, Driver) ? ClassType.ValueType : @class.Type;
+            return base.VisitClassDecl(@class);
+        }
+
+        private bool CheckClassIsStructible(Class @class, Driver Driver)
+        {
+            if (@class.IsValueType)
+                return true;
+
+            if (!@class.IsStruct)
```

Comment:
Me and @ddobrev discussed this and we think it's better to replace all this logic with `@class.IsPOD`.